test/simulator/misc/random-teams.js

```diff
@@ -128,13 +138,27 @@ describe(`Factory sets`, function () {
 							assert(ability.name === abilityName, `miscapitalized ability "${abilityName}" of ${species}`);
 						}
 
+						for (const natureName of [].concat(set.nature)) {
```

Comment:
what is going on here? it would be better as

```javascript
for (const natureName of [set.nature]) { ... }
```

But you don't even need the loop at all if theres only one element! (and if I'm missing something and `set.nature` is already an array, just loop over that instead of the `concat`).

Reply:
set.nature can be an array or a string, and i just did this because it was consistent with the other loops

Reply:
I see. I think that they can be written more cleanly as `[...set.nature]` then :)

Reply:
That would result in Nature Jolly becoming ``['J', 'o', 'l', 'l', 'y']``.
``Array.prototype.concat()`` is special in that it accepts as arguments single elements to be pushed, but it also flattens arrays with ``Symbol.isConcatSpreadable``.

Reply:
TIL, thanks for the explanation.

Reply:
Considering how obscure this is, it should probably be commented somewhere.